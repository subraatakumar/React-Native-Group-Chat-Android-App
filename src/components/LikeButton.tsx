import React from 'react';
import {View} from 'react-native';
import {increaseLike, useAppDispatch} from '../redux/store';
import {ThemeColors} from '../settings/config';
import {SingleChatMessageType} from '../settings/types';
import CustomButton from './CustomButton';

type PositionTypes = 'left' | 'right';

type LikeButtonPropTypes = {
  item: SingleChatMessageType;
  uid: string;
  disabled?: boolean;
  tc?: string; // Text Color
  position?: PositionTypes;
  value?: number;
  setValue?: Function;
};
const LikeButton = ({
  item,
  uid,
  value = 0,
  setValue,
  disabled = false,
  tc = disabled ? ThemeColors.disabled : ThemeColors.primary,
  position = 'left',
}: LikeButtonPropTypes) => {
  //const likes = item.likes.length;
  const likes = item && item?.likes ? item?.likes.length : 0;
  const dispatch = useAppDispatch();

  const increaseLikeByOne = () => {
    if (!item.likes?.includes(uid) && !disabled) {
      dispatch(increaseLike({item, uid}));
    }
  };

  return (
    <View style={{position: 'absolute', bottom: -15, [position]: 10}}>
      <CustomButton
        leftIcon={'thumbs-up-outline'}
        title={'' + likes}
        w={'auto'}
        bgc={'#fff'}
        tc={tc}
        ts={12}
        leftIconSize={10}
        m={0}
        p={2}
        onPressFn={increaseLikeByOne}
      />
    </View>
  );
};

export default LikeButton;
