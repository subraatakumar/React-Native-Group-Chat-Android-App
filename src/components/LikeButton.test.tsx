import React from 'react';
import renderer from 'react-test-renderer';
import LikeButton from './LikeButton';

test('Like Button renders correctly', () => {
  const tree = renderer
    .create(
      <LikeButton
        item={{
          docId: '',
          text: '',
          createdAt: undefined,
          sentId: '',
          receivedId: '',
          groupId: null,
          likes: null,
        }}
        uid={''}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
