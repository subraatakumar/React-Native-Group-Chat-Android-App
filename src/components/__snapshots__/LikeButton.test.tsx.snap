// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Like Button renders correctly 1`] = `
<View
  style={
    Object {
      "bottom": -15,
      "left": 10,
      "position": "absolute",
    }
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 30,
        "elevation": 8,
        "height": null,
        "margin": 0,
        "opacity": 1,
        "paddingBottom": 2,
        "paddingLeft": 2,
        "paddingRight": 2,
        "paddingTop": 2,
        "shadowColor": "rgba(6,146,239,1)",
        "shadowOffset": Object {
          "height": 5,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 8,
        "width": "auto",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "auto",
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "rgba(0,128,105,1)",
              "fontSize": 10,
            },
            Object {
              "marginLeft": 2,
              "marginRight": 3,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ï•¯
      </Text>
      <Text
        style={
          Object {
            "color": "rgba(0,128,105,1)",
            "fontSize": 12,
            "textAlign": "center",
            "width": "auto",
          }
        }
      >
        0
      </Text>
      <View
        style={
          Object {
            "marginLeft": 2,
            "marginRight": 2,
          }
        }
      />
    </View>
  </View>
</View>
`;
